pkg/planner/core/rule_column_pruning.go

```diff
@@ -414,6 +416,15 @@ func (ds *DataSource) PruneColumns(parentUsedCols []*expression.Column, opt *log
 	if ds.handleCols != nil && ds.handleCols.IsInt() && ds.schema.ColumnIndex(ds.handleCols.GetCol(0)) == -1 {
 		ds.handleCols = nil
 	}
+	if !addOneHandle && ds.schema.Len() > len(parentUsedCols) && ds.SCtx().GetSessionVars().IsMPPEnforced() {
+		proj := LogicalProjection{
+			Exprs: expression.Column2Exprs(parentUsedCols),
+		}.Init(ds.SCtx(), ds.QueryBlockOffset())
+		proj.SetStats(ds.StatsInfo())
+		proj.SetSchema(expression.NewSchema(parentUsedCols...))
+		proj.SetChildren(ds)
+		return proj, nil
+	}
```

Comment:
It's not a very good idea to add it here. Let me take a deeper look.

Reply:
Please let me know if any suggestions.

Reply:
> It's not a very good idea to add it here. Let me take a deeper look.

1. I think this is not a only mpp task enhancement but also suite for tikv task
2. Should we change the output schema of datasource rather than add a projection operator in here ?

Reply:
Yeah, for point 1, tikv task will benefit from this if projection push down work, and this will be the next work I'll focus on(will see more detail in [issue](https://github.com/pingcap/tidb/issues/52133)), currently, limit it to mpp task only.
For point 2, as I know, we can't change the output schema here. Please correct me if I'm wrong @winoros